src/ui/qgsjoindialogbase.ui

```diff
@@ -126,6 +126,13 @@
      </property>
     </widget>
    </item>
+   <item row="5" column="0">
+    <widget class="QCheckBox" name="mDynamicFormCheckBox">
+     <property name="text">
+      <string>Dynamic form</string>
```

Comment:
Can you add a tooltip? I would have guessed that this would make fields updatable. (my fault but just to say that the name wasn't totally obvious to me)